### PR TITLE
Hotfix for RRTMGP using multiple threads

### PR DESCRIPTION
**Pull request by dustinswales**
This PR contains several changes to the initialization and setup of the RRTMGP radiation scheme to allow for use across multiple openMP threads.
*) Moved Interstitial RRTMGP DDTs (ty_rrtmgp_gas_optics, ty_cloud_optics) to static fields defined during initialization in module memory.
*) Add "$omp critical" statements around the calling type-bound procedures during initialization.
*) Move allocation of ty_gas_conc from Interstitial to GFS_typedefs. Initialize ty_gas_concs for all blocks during initialization.

This issue was brought to our attention by @yangfanglin and Qingfu Liu at EMC while testing RRTMGP in high-resolution cases, which require multiple threads.

The GP regression tests on hera using Intel were successful. 
There are no changes to the baselines

Waiting on https://github.com/NCAR/ccpp-physics/pull/568

### CODE REVIEW

**Comment by yangfanglin** [comment]
Please add Qinfu Liu as a reviewer

**Review by junwang-noaa — APPROVED** [review]

**Comment by Qingfu-Liu** [comment]
Dusan,
    Thank you very much. I will start reviewing the code changes.
Qingfu

**Review by Qingfu-Liu — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

File: ccpp/data/GFS_typedefs.F90 [hunk 1]
```diff
@@ -6438,6 +6437,14 @@ subroutine interstitial_create (Interstitial, IM, Model)
        allocate (Interstitial%toa_src_sw           (IM,Model%rrtmgp_nGptsSW))
        allocate (Interstitial%toa_src_lw           (IM,Model%rrtmgp_nGptsLW))
        allocate (Interstitial%active_gases_array   (Model%nGases))
+       ! ty_gas_concs
```
Comment (climbfuji) [inline ccpp/data/GFS_typedefs.F90:6440]: Obviously %ncol and %nlay are fixed throughout the run.

What about gas_name, concs, conc? Do they depend on each grid, and if yes, do they need to be persistent? If not, then we should add them to the %rad_reset calls for the DDT (for clarity) - in the next PR. If they need to be persistent, they need to be moved to radtend or something like that (also in the next PR).

**Review by climbfuji — APPROVED** [review]

File: ccpp/data/GFS_typedefs.F90 (hunk 1 above)
Comment (dustinswales) [reply ccpp/data/GFS_typedefs.F90:6440]: @climbfuji 
I set the ncol and nlay here once in the DDT, type-bound procedures rely on it later.
gas_concentrations(ty_gas_concs) should be added to rad_reset. I will make sure to include this in my next PR.

**Pull request merged**